src/node_crypto.cc

```diff
@@ -1974,7 +1974,78 @@ static Local<Object> X509ToObject(Environment* env, X509* cert) {
 }
 
 
-// TODO(indutny): Split it into multiple smaller functions
+// Put issuer inside the object
+static void AddIssuer(X509** cert,
+                      const STACK_OF(X509)* const peer_certs,
+                      Local<Object>* const info,
+                      Environment* const env) {
+  *cert = sk_X509_delete(peer_certs, 0);
+  while (sk_X509_num(peer_certs) > 0) {
+    int i;
+    for (i = 0; i < sk_X509_num(peer_certs); i++) {
+      X509* ca = sk_X509_value(peer_certs, i);
+      if (X509_check_issued(ca, *cert) != X509_V_OK)
+        continue;
+
+      Local<Object> ca_info = X509ToObject(env, ca);
+      (*info)->Set(env->issuercert_string(), ca_info);
+      *info = ca_info;
+
+      // NOTE: Intentionally freeing cert that is not used anymore
```

Comment:
Since you're here, can you punctuate the comments?

Reply:
Absolutely.